react/LocationInput.tsx

```diff
@@ -39,14 +42,39 @@ const LocationInput: React.FC<Props> = ({
 
   const countryRules = country ? rules[country] : undefined
 
+  const formatAndValidate = () => {
+    const postalCodeMask = countryRules?.fields.postalCode?.mask
+
+    if (!postalCodeMask) {
+      return
+    }
+
+    setInputValue(msk(inputValue, postalCodeMask))
+
+    const patternInvalid =
+      inputValue.length &&
+      countryRules?.fields.postalCode?.pattern != null &&
+      inputValue.match(countryRules.fields.postalCode.pattern) === null
+
+    if (patternInvalid) {
+      setInvalidPostalCode(true)
+    }
+  }
+
+  useEffect(() => {
+    setInvalidPostalCode(false)
+    formatAndValidate()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [countryRules])
```

Comment:
why disabling the exhaustive deps here? if you remove this comment and use `useCallback` in the `formatAndValidate` function you would get the same behavior (because the function actually depends on `countryRules`)

Reply:
the thing is, with `useCallback` and adding `formatAndValidate` to the deps of `useEffect` results in the effect being triggered everytime `formatAndValidate` gets called. is there any other alternative?

Reply:
I see, that's because `formatAndValidate` depends on `inputValue`, right? you could update it to instead of using the value from the closure to use it from an argument, then you wouldn't have this infinite loop

Reply:
ohh that's right. I'll change it

Reply:
but wait, now the useEffect depends on `inputValue` in order to send it to `formatAndValidate`. Am I missing something?

Reply:
here is my attempt

```typescript
  const formatAndValidate = useCallback(
    (postalCode: string) => {
      // other stuff here
    },
    [
      countryRules?.fields.postalCode?.mask,
      countryRules?.fields.postalCode?.pattern,
    ]
  )

  useEffect(() => {
    setInvalidPostalCode(false)
    formatAndValidate(inputValue) // <- how I should handle this?
  }, [countryRules, formatAndValidate])
```

Reply:
you can use a ref to only trigger the effect when the `countryRules` dependency changes, e.g:

```js
const formatAndValidate = useCallback(/* ... */)

const prevCountryRulesRef = useRef(
  null // Use null if you want the effect to trigger on first render, else use `countryRules`
)

useEffect(() => {
  if (prevCountryRulesRef.current === countryRules) {
    return
  }

  prevCountryRulesRef.current = countryRules

  setInvalidPostalCode(false)
  formatAndValidate(inputValue)
}, [
/* dependencies doesn't matter anymore, since you are
 * now only running the effect when there is a change to
 * the countryRules ref, so you can safely include any deps
 * the eslint rule wants to
 */
])
```